### ISSUE TITLE
Inputs in shap.DeepExplainer.shap_values(X), where X does not contains only PyTorch tensors

### ISSUE THREAD

**Issue by gionanide** [shap/shap#2080]
Hello,
First of all I want to thank you for your contribution and the repository.

Currently I am facing a problem using  shap.DeepExplainer.shap_values(X).
My X is as follows ([x_1,x_2,x_3]), where
1. x_1 is  PyTorch tensor
2. x_2 is a python list
3. x_3 is a PyTorch nn.Module

Given this input I am getting an error in line 143 (deep_pytorch.py)
```python
X = [x.detach().to(self.device) for x in X]
```
which is not letting me pass other type of inputs besides tensors.

I tried to overcome this by setting torch.tensor(x_2), but I can not do the same with x_3 which is a nn.Module object.

Thus, the following code maybe helps which discriminates between tensors or other types of objects before apply .detach().

```python
for i in range(len(X)):
  if(torch.is_tensor(X[i])):
    X[i] = X[i].detach().to(self.device)
```
